bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5182,6 +5187,17 @@ void ReplicaImp::onExecutionFinish() {
     // We are about to stop execution. To avoid VC we now clear all pending requests
     clientsManager->clearAllPendingRequests();
   }
+
+  // Sending noop client requests to get the system into a stable checkpoint,
+  // allowing the create dbCheckpoint operator command to create a dbCheckpoint/snapshot.
+  if (getReplicaConfig().dbCheckpointFeatureEnabled && isCurrentPrimary() &&
```

Comment:
Can we do this:

 if (getReplicaConfig().dbCheckpointFeatureEnabled && isCurrentPrimary() &&
      (DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot().has_value() &&
       DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot().value() >= lastExecutedSeqNum)) {

       bool noop = (DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot().value() == lastExecutedSeqNum+1) ? false : true;    DbCheckpointManager::instance().sendInternalCreateDbCheckpointMsg(
        lastExecutedSeqNum + 1, noop);  // send noop till we reach stable checkpoint
       if(!noop) DbCheckpointManager::instance().setNextStableSeqNumToCreateSnapshot(std::nullopt);
       LOG_INFO(GL, "sendInternalCreateDbCheckpointMsg noop command ");
  }

With this, we do not need to modify the callback for creating DbCheckpoints.

Reply:
Fixed